std/digest/digest.d

```diff
@@ -1021,3 +1021,53 @@ if (isDigest!T) : Digest
     ubyte[5] buf;
     assert(hash.peek(buf).toHexString() == "39A34F41");
 }
+
+/**
+ * Securely compares two digest representations. This function is $(BIGO n) for all
+ * strings that are of equal length. Use of this function is recommended
+ * for comparing verification tokens, as it protects from brute force timing
+ * attacks that can reveal the key.
+ *
+ * Params:
+ *     s1 = A digest representation
+ *     s2 = A digest representation
+ * Returns:
+ *     `true` if both strings are equal, `false` otherwise
+ * See_Also:
+ *     $(LINK2 https://security.stackexchange.com/a/74552, An in depth explanation of
+ *     the attack).
+ */
+bool secureCompare(T)(T[] s1, T[] s2)
+{
+    if (s1.length != s2.length)
+    {
+        return false;
+    }
+
+    int result = 0;
+    foreach (i; 0 .. s1.length)
+    {
+        result |= s1[i] ^ s2[i];
```

Comment:
This requires a sig constraint that `T` (1) supports the `^` operation, (2) that said operation does what we think it does, and (2) the result can be OR'd into an `int`.